lib/githubbot.ts

```diff
@@ -0,0 +1,403 @@
+/*
+Copyright 2016 Resin.io
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import * as ProcBot from './procbot';
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import * as fs from 'fs';
+import * as jwt from 'jsonwebtoken';
+import { AlertLevel } from './procbot';
+
+// The GithubAPI Promise implies built-in ES6 Promises. Therefore when assigning
+// the Promise to use as Bluebird, VS Code reports an error. Need a good
+// fix for this that doesn't require type declarations?
+const GithubApi = require('github');
+const request: any = Promise.promisifyAll(require('request'));
```

Comment:
https://www.npmjs.com/package/request-promise - Bluebird built in (and other builds available for diff Promises if you like), with type definitions published too.